composable.el

```diff
@@ -238,24 +234,24 @@ For each function named foo a function name composable-foo is created."
   (set-transient-map
    (composable--singleton-map
     (vector last-command-event)
-    (composable--repeater composable--command object
-			  (composable--direction last-prefix-arg)))
+    (composable--repeater composable--command object last-prefix-arg))
    t
-   'composable--object-exit))
+   #'composable--object-exit))
 
 (defun composable--handle-prefix (command)
   "Handle prefix arg where the COMMAND is paired in PAIRS."
-  (let ((pair (gethash command composable--fn-pairs)))
+  (let ((pair (or (alist-get command composable-fn-pair-alist)
```

Comment:
The old approach of using a hash map seems cleaner to me, and it is more efficient.

Reply:
Hi:

The old approach inserted duplicated elements in the list which is somehow unneeded store '(a b) and '(b a) is redundant and in case we want to distinguish open from close commands is more complex to discriminate. With this, in the future, we can distinguish searching in the forwards `(alist-get command composable-fn-pair-alist)` from searching in the backward `(car (rassq command composable-fn-pair-alist))` if we need that.

But also the customization with the alist is simpler from the user point of view. Because there is not magic function needed; just need to add pairs to the alist. So, this eliminates the need of an extra function to add pairs and exposes this customization variable directly to the user more in the `emacs style` and the user could use either setq, the customize-variable interface, add-to-list and so on.

Finally using a hash table for small arrays like this, is not more efficient than using an alist. Even for big sets it is not so simple and has been discussed many times in the mailing list

https://lists.nongnu.org/archive/html/bug-gnu-emacs/2019-10/msg00779.html 

The hash table overhead is so big that the performance benefit only exists (sometimes) when the set is in the order of the thousands.

Reply:
I see, that makes sense. Thank you for the explanation.